ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala

```diff
@@ -48,32 +53,130 @@ private class PackageService(reloadPackageStoreIfChanged: PackageService.ReloadP
   @volatile private var state: State =
     State(Set.empty, TemplateIdMap.Empty, Map.empty, Map.empty, Map.empty)
 
-  // synchronized, so two threads cannot reload it concurrently
-  def reload(implicit
-      ec: ExecutionContext,
-      lc: LoggingContextOf[InstanceUUID],
-  ): Future[Error \/ Unit] =
-    synchronized {
-      reloadPackageStoreIfChanged(state.packageIds).map {
-        _.map {
-          case Some(diff) =>
-            this.state = this.state.append(diff)
-            logger.info(s"new package IDs loaded: ${diff.keySet.mkString(", ")}")
-            logger.debug(s"loaded diff: $diff")
-            ()
-          case None =>
-            logger.debug(s"new package IDs not found")
-            ()
+  // volatile, because we want to reduce the probability of unnecessary package updates
+  // due to multiple threads reading this.
+  // Writes to this are synchronized with a mutex, so in theory this should work out fine.
+  @volatile private var lastUpdated = LocalDateTime.MIN
+
+  // Regular updates should happen regardless of the current state every minute.
+  private def packagesShouldBeFetchedAgain =
+    lastUpdated.until(LocalDateTime.now(), temporal.ChronoUnit.SECONDS) >= 60L
+
+  private val _reload = {
+    // Because things can go so easily wrong, the declaration of the mutex
+    // was moved into a separate block, so it's easier to track down all interactions.
+    val packageUpdateSync = util.Concurrent.Mutex()
```

Comment:
so if I read this correctly the way the concurrency logic works is as follows:
`updateTaskUpateSync` synchronizes updates to the `ongoingPackageUpdateTask` variable.
`updateTaskUpdateSync` is locked while the future executes. I don’t actually understand why this needs to be a mutex.

I agree with @stefanobaghino-da that we can simplify that a bit.